File: library/src/main/java/com/silverhetch/aura/view/measures/ScreenSize.java
package com.silverhetch.aura.view.measures;

import android.content.Context;
import android.graphics.Point;
import android.view.Display;
import android.view.WindowManager;

/**
 * Created by larryhsiao on 2017/10/23.
 */

public class ScreenSize {
    private final Point size;

    public ScreenSize(Context context) {
        this.size = new Point();
        WindowManager wm = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
        Display display = wm.getDefaultDisplay();
        display.getSize(size);
    }

    public int width() {
        return size.x;
    }

    public int height() {
        return size.y;
    }
}

File: library/src/main/java/com/larryhsiao/aura/view/measures/ScreenSize.kt
package com.larryhsiao.aura.view.measures

import android.content.Context
import android.graphics.Point
import android.view.WindowManager
import com.larryhsiao.clotho.Source

/**
 * Utility to grant the screen size from context.
 */
class ScreenSize(context: Context) : Source<Point> {
    private val size: Point by lazy {
        Point().apply {
            val wm = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
            val display = wm.defaultDisplay
            display.getSize(this)
            this
        }
    }

    override fun value(): Point {
        return size
    }

    fun width(): Int {
        return size.x
    }

    fun height(): Int {
        return size.y
    }
}
